lib/pbench/server/cache_manager.py

```diff
@@ -107,28 +107,31 @@ def __init__(self, dir_name: str, path: Path):
         else:
             self.location = Path(dir_name) / str(path).split(f"{dir_name}/")[1]
 
-        try:
-            resolve_path = path.resolve(strict=True)
-        except FileNotFoundError:
-            raise FileNotFoundError(f"The '{path}' does not resolve to a real location")
-
-        if resolve_path.name == dir_name:
-            self.resolve_path = dir_name
-        else:
-            self.resolve_path = (
-                Path(dir_name) / str(resolve_path).split(f"{dir_name}/")[1]
-            )
-
+        self.resolve_path = None
         self.resolve_type = None
         self.size = None
         if path.is_symlink():
             self.type = "symlink"
-            self.resolve_type = "directory" if resolve_path.is_dir() else "file"
+            try:
+                resolve_path = path.resolve(strict=True)
+                self.resolve_type = "directory" if resolve_path.is_dir() else "file"
+            except FileNotFoundError:
+                resolve_path = path.readlink()
+                self.resolve_type = "symlink"
+
+            if resolve_path.name == dir_name:
+                self.resolve_path = dir_name
+            else:
+                self.resolve_path = (
+                    Path(dir_name) / str(resolve_path).split(f"{dir_name}/")[1]
+                )
         elif path.is_file():
             self.type = "file"
             self.size = path.stat().st_size
         elif path.is_dir():
             self.type = "directory"
+        else:
+            self.type = "other"
```

Comment:
I wonder what "other" types we can actually get out of a tarball? I suppose in theory we could have a socket, a mount point, a fifo ... but I tend to agree it might not be worthwhile breaking them out. (Then again, `stat` ought to give us the details, and maybe we should report that directly?)

Reply:
I had the reverse thought:  I'd rather not see "other"-type files unpacked at all.  (I don't know what creating them might do...and I don't want to contemplate what trying to fetch them for an inventory request might do!)  So, perhaps this is a problem that we could "prevent" instead of having to cope with....

Reply:
We're not talking about "unpacking" ... we're talking about "mapping". There's no point in "unpacking" (e.g., via `/datasets/<id>/inventory/<path>`) a socket or fifo; there's no conceivably useful information there at the point of extraction. (I'm not even sure what byte stream we'd get, since there's clearly no archived connection to a kernel IO structure.) This should probably be a 4xx error, maybe something like `UNSUPPORTED_MEDIA_TYPE` (415)?

However, if a client is trying to delve into the structure of the captured dataset with `/datasets/<id>/contents/<path>`, and there's a `fifo` or `socket` or mount point captured in the tarball ... yeah, I'd like the client to know that "at this path there's a <whatever>". I'm not thrilled about "other", and hiding it completely is worse. We *can* use the `stat` bits or the `is_*` Python functions to decode the type, and I kinda think we should.

Reply:
In the cited code above, we are assessing the contents of the result tarball as unpacked for us by the previous stage in our upload pipeline (the "indexer"?...whatever) so that we can build a map of it.  If we can assert that that stage declined to create any "weird" file-objects (like sockets or pipes), then we're in good shape:  we don't have to worry about typing them in this code, and the inventory code doesn't have to worry about what will happen if it tries to read from them.

Nevertheless, if you really want a user to be able to look at the map and see that there is a "thing" at that path, then, um, yeah, the unpacker needs to put _something_ at that location, and the map-builder needs to recognize it for what it is/was-originally, and we need to make sure that the inventory API doesn't try to actually read from it.

But, I'm not sure that we want to let `tar` create actual sockets, FIFOs, or block- or character-device files in our file system, just because some user managed to get them into the tarball.  (Sure, that would let `stat` do the heavy lifting for us, but it seems like a bad idea overall.)